packages/snackbar/index.tsx

```diff
@@ -153,6 +153,14 @@ export class Snackbar extends React.Component<Props, State> {
       this.foundation.open();
     }
   }
+  componentDidUpdate(prevProps: Props) {
+    const {open} = this.props;
+    if (prevProps.open !== open) {
+      if (open) {
+        this.foundation.open();
+      }
```

Comment:
shouldn't you call `this.foundation.close()` on an else clause?

Reply:
Correct...we need the `this.foundation.close(reason)`. I suspect we would also want a `closedReason` prop that gets passed to `reason` in the close method. I'm open to other ideas too.

Reply:
we could use the prop with a default of the dismissed value from the mdc snackbar, what do you think?

Reply:
[this one](https://github.com/material-components/material-components-web/blob/master/packages/mdc-snackbar/constants.ts#L40).

Reply:
@lucasecdb I don't think we need to call `this.foundation.close()` because the `foundation` is aware when the snackbar is closed. So why not just utilize the `notifyClosed` method of the `adapter` and use the `onClose` prop to update the component's `isOpen` state. Something like this:
```js
const Component = () => {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <Snackbar open={isOpen} onClose={() => setIsOpen(false)} />
  )
}
```
This method is from the `Dialog` component and I think this makes sense.

Reply:
oh, right. but what if the user set the `open` to false _before_ the foundation actually auto-closes the snackbar?

Reply:
@lucasecdb sorry, my bad. Now I understand what you mean.

Reply:
The use case is if someone wants to programmatically close the snackbar before it automatically closes. I've attached a [patch](https://stackoverflow.com/questions/5159185/create-a-git-patch-from-the-changes-in-the-current-working-directory/15438863#15438863) to add on to the screenshot tests. 

[snackbar.patch.zip](https://github.com/material-components/material-components-web-react/files/2937257/snackbar.patch.zip)

Reply:
@moog16 I've applied the patch. I didn't event know this thing exists. thanks :)
I just called foundation.close with an empty argument if open prop is false.

Reply:
Can you add another prop for the reason? I'm thinking something like:
```
const {reason} = this.props;
...
this.foundation.close(reason)
```

Reply:
@moog16 Sure. Sorry for the late replies. I'll push the update you need in a moment.